awx/main/tasks.py

```diff
@@ -372,16 +389,30 @@ def gather_analytics():
             if acquired is False:
                 logger.debug('Not gathering analytics, another task holds lock')
                 return
-            try:
-                tgz = analytics.gather()
-                if not tgz:
-                    return
-                logger.info('gathered analytics: {}'.format(tgz))
-                analytics.ship(tgz)
+            subset = settings.AUTOMATION_ANALYTICS_COLLECTOR_SUBSET or list(all_collectors().keys())
+            incremental_collectors = []
+            for collector in expensive_collectors():
+                if collector in subset:
+                    subset.remove(collector)
+                    incremental_collectors.append(collector)
+
+            # Cap gathering at 4 weeks of data if there has been no data gathering
+            since = last_time or (gather_time - timedelta(weeks=4))
+
+            errors = 0
+            if subset:
+                errors += _gather_and_ship(subset, since=since, until=gather_time)
+            for collector in incremental_collectors:
+                start = since
+                until = None
+                while start < gather_time:
+                    until = start + timedelta(minutes=20)
+                    if (until > gather_time):
+                        until = gather_time
+                    errors += _gather_and_ship(collector, since=start, until=until)
```

Comment:
@wenottingham just to make sure, do we always send `config.json` with each payload? I got lost in debugging of the path :-)

config.json is required, since all data saved is tied to cluster_id we get from this

Reply:
No, it doesn't guarantee that at the moment, will need fixed.

Reply:
Maybe `manifest.json` too, we don't use the versions anywhere now, but each payload should probably have it just for sure, what do you think @benthomasson ?

Reply:
manifest is always added, config.json would go in the same block